src/main/java/io/cdap/e2e/pages/locators/CdfConnectionLocators.java

```diff
@@ -50,4 +51,20 @@ public static By locatorOfDataTableTitle(String title) {
   public static By locatorOfPipelineTypeButton(String pipelineType) {
     return By.xpath("//a//*[text()='" + pipelineType + "']");
   }
+
+  public static WebElement locateConnectionType(String connectionType) {
+    return SeleniumDriver.getDriver().findElement(
+      By.xpath("//div[@data-cy='categorized-connection-type-" + connectionType + "']"));
+  }
+
+  public static WebElement locateConnectionActionMenu(String connectionName) {
+    return SeleniumDriver.getDriver().findElement(By.xpath("//*[text()='" + connectionName + "']" +
```

Comment:
Can we use `div[@data-cy='connection-connectionType-connectionName']` instead of `text()`?

Reply:
Updated locator with data-cy